common/djangoapps/entitlements/tasks.py

```diff
@@ -62,3 +72,49 @@ def expire_old_entitlements(self, start, end, logid='...'):
         raise self.retry(exc=exc, countdown=countdown, max_retries=MAX_RETRIES)
 
     LOGGER.info('Successfully completed the task expire_old_entitlements after examining %d entries [%s]', entitlements.count(), logid)  # lint-amnesty, pylint: disable=line-too-long
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def expire_and_create_entitlements(self, no_of_entitlements):
+    """
+    This task is designed to be called to process and expire bundle of entitlements
+    that are older than one year on in exceptional case 18 months.
+
+    Args:
+        None
+
+    Returns:
+        None
+
+    """
+    LOGGER.info('Running task expire_and_create_entitlements')
+
+    current_date = date.today()
+    expiration_period = current_date - relativedelta(years=1)
+    exceptional_expiration_period = current_date - relativedelta(years=1, months=6)
+    normal_entitlements = CourseEntitlement.objects.filter(
+        expired_at__isnull=True, created__lte=expiration_period).exclude(course_uuid__in=MIT_SUPPLY_CHAIN_COURSES)
+    exceptional_entitlements = CourseEntitlement.objects.filter(
+        expired_at__isnull=True, created__lte=exceptional_expiration_period, course_uuid__in=MIT_SUPPLY_CHAIN_COURSES)
+
+    entitlements = normal_entitlements | exceptional_entitlements
+    countdown = 2 ** self.request.retries
+
+    try:
+        for entitlement in entitlements[:no_of_entitlements]:
+
+            entitlement.expire_entitlement()
```

Comment:
Will we be able to identify which entitlements were expired as part of this work?  I worry they may not be distinguishable except maybe by parsing logs.  Perhaps it would be good to replicate the support-tools method and add a description so we can see in the DB (and in support-tools) the reason why the entitlement was expired.